src/main/scala/com/exasol/cloudetl/parquet/converter/ParquetConverter.scala

```diff
@@ -134,3 +140,219 @@ final case class ParquetDateConverter(index: Int, holder: ValueHolder)
+  val parentDataHolder: ValueHolder
+) extends GroupConverter
+    with ParquetConverter
+    with ArrayConverter {
+
+  override def createElementConverter(): Converter = new GroupConverter {
+    val innerConverter = ConverterFactory(elementType, index, dataHolder)
+    override def getConverter(index: Int): Converter = innerConverter
+    override def start(): Unit = {}
+    override def end(): Unit = {}
+  }
+}
+
+final case class MapConverter(groupType: GroupType, index: Int, parentDataHolder: ValueHolder)
+    extends GroupConverter
+    with ParquetConverter {
+  private[this] val keysDataHolder = new AppendedValueHolder()
+  private[this] val valuesDataHolder = new AppendedValueHolder()
+  private[this] val converter = createMapConverter()
+
+  override def getConverter(fieldIndex: Int): Converter = {
+    if (fieldIndex < 0 || fieldIndex > 1) {
+      throw new IllegalArgumentException(
+        s"Illegal index '$fieldIndex' to map converter. It should be " +
+          "either '0' for keys converter or '1' for values converter."
+      )
+    }
+    converter
+  }
+  override def start(): Unit = {
+    keysDataHolder.reset()
+    valuesDataHolder.reset()
+  }
+  override def end(): Unit = {
+    val keys = keysDataHolder.getValues()
+    val values = valuesDataHolder.getValues()
+    val map = keys.zip(values).toMap
+    parentDataHolder.put(index, map)
+  }
+
+  private[this] def createMapConverter(): Converter = new GroupConverter {
+    val mapType = groupType.getFields().get(0).asGroupType()
+    val mapKeyType = mapType.getFields().get(0)
+    val mapValueType = mapType.getFields().get(1)
+    val keysConverter = ConverterFactory(mapKeyType, index, keysDataHolder)
+    val valuesConverter = ConverterFactory(mapValueType, index, valuesDataHolder)
+
+    override def getConverter(index: Int): Converter =
+      if (index == 0) {
+        keysConverter
+      } else {
+        valuesConverter
+      }
+    override def start(): Unit = {}
+    override def end(): Unit = {}
+  }
+}
+
+@SuppressWarnings(Array("org.wartremover.contrib.warts.UnsafeInheritance"))
+class AbstractStructConverter(groupType: GroupType, index: Int, parentDataHolder: ValueHolder)
+    extends GroupConverter {
+  private[this] val size = groupType.getFieldCount()
+  protected[this] val dataHolder = IndexedValueHolder(size)
+  private[this] val converters = createFieldConverters()
+
+  override def getConverter(fieldIndex: Int): Converter = converters(fieldIndex)
+  override def start(): Unit = dataHolder.reset()
+  override def end(): Unit = parentDataHolder.put(index, dataHolder.getValues())
+
+  private[this] def createFieldConverters(): Array[Converter] = {
+    val converters = Array.ofDim[Converter](size)
+    for { i <- 0 until size } {
+      converters(i) = ConverterFactory(groupType.getType(i), i, dataHolder)
+    }
+    converters
+  }
+}
+
+final case class StructConverter(groupType: GroupType, index: Int, parentDataHolder: ValueHolder)
```

Comment:
add doc to all converters. I don't get the different purposes

Reply:
Fixed.